### ISSUE TITLE
Cloud connector

### ISSUE THREAD

**Issue by ktzoumas** [stratosphere/stratosphere#40]
We need a cloud connector to ozone.

Why? This is needed already by one partner (Deutsche Telekom), and I expect that more companies organize their clusters as private clouds.

We should create the minimum possible API glue without fancy stuff for now and leave the smart stuff to the outside. We can probably reuse and refactor the ec2-cloudmanager from Stratosphere v0.2. 

Ideally, we should cover Amazon EC2 and OpenStack (DT controls their cluster with Vagrant) with one stone

**Comment by StephanEwen** [comment]
What exactly will the cloud connector do? Will it bring up Stratosphere instances on a cloud infrastructure? In that case we should be able to reuse much of the EC2 manager.

**Comment by ktzoumas** [comment]
I think the functionality will be similar. Should I close this until I get more specific requirements?

**Comment by rmetzger** [comment]
I totally agree on this issue. We need automated deployment in cloud infrastructures.

I think the issue is fixable within one day. Its only a matter of api's and some automated configuration, we don't have to change anything on the code.

These projects could be helpful.
http://whirr.apache.org/
http://jclouds.incubator.apache.org/

Hadoop has ec2 support since years.

I'd like to test stratosphere automatically in such an infrastructure on a regular basis.

**Comment by uce** [comment]
This is on my agenda.

**Comment by rmetzger** [comment]
Lets have a meeting about this

Sent from my iPhone

On 27.10.2013, at 02:04, Ufuk Celebi notifications@github.com wrote:

This is on my agenda.

—
Reply to this email directly or view it on
GitHubhttps://github.com/stratosphere/stratosphere/issues/40#issuecomment-27158876
.

**Comment by rmetzger** [comment]
@tobwiens is working on this: https://github.com/tobwiens/stratDeployEc2
